child_compassion/models/child_compassion.py

```diff
@@ -441,8 +441,16 @@ def child_released(self):
             'sponsor_id': False,
             'state': 'R'
         })
+
         other_children = self - sponsored_children
-        other_children._postpone_deletion()
+        other_children.get_lifecycle_event()
+
+        # postpone the deletion of children which haven't been sponsored
+        postpone = datetime.now() + timedelta(weeks=1)

```

Comment:
Instead of puting "one week" which is static, I just had a better idea. Let's take the hold expiration date as the time where we will remove this child. To retrieve it you can search for the hold.
```
for child in other_children:
    hold = self.env['compassion.hold'].search([('child_id', '=', child.id)])
    postpone = fields.Datetime.from_string(hold.expiration_date)
```